doc/modules/grid_search.rst

```diff
@@ -150,6 +154,324 @@ increasing ``n_iter`` will always lead to a finer search.
+
+.. _amount_of_resource_and_number_of_candidates:
+
+Amount of resource and number of candidates at each iteration
+-------------------------------------------------------------
+
+The amount of resources ``resource_iter`` (e.g. the number of samples) allocated for
+each candidate at iteration ``i`` is controlled by the parameters ``ratio``
+and ``min_resources`` as follows::
+
+    resource_iter = ratio**i * min_resources
+
+``min_resources`` is the amount of resources used at the first iteration and
+``ratio`` defines the proportions of candidates that will be selected for
+the next iteration::
+
+    n_candidates_to_keep = n_candidates_at_i // ratio
+
+So in the first iteration, we use ``min_resources`` resources ``n_candidates``
+times. In the second iteration, we use ``min_resources * ratio`` resources
+``n_candidates // ratio`` times. The third again multiplies the resources
+per candidate and divides the number of candidates. This process stops when
+the maximum amount of resource per candidate is reached, or when less than
+``ratio`` candidates are left.
+
+Here is an example with ``min_resources=3`` and ``ratio=2``, starting with 70
+candidates:
+
++-------------+-----------------------+
+| ``resource_iter``     | ``n_candidates_at_i`` |
++=============+=======================+
+| 3 (=min_resources)  | 70 (=n_candidates)    |
++-------------+-----------------------+
+| 3 * 2 = 6   | 70 // 2 = 35          |
++-------------+-----------------------+
+| 6 * 2 = 12  | 35 // 2 = 17          |
++-------------+-----------------------+
+| 12 * 2 = 24 | 17 // 2 = 8           |
++-------------+-----------------------+
+| 24 * 2 = 48 | 8 // 2 = 4            |
++-------------+-----------------------+
+| 48 * 2 = 96 | 4 // 2 = 2            |
++-------------+-----------------------+
+
+Ideally, at the last iteration, ``ratio`` candidates are evaluated, and we
+can pick the best one. Note that each ``resource_iter`` is a multiple of both
+``ratio`` and ``min_resources``.
+
+The amount of resource that is used at each iteration can be found using the
+`cv_results_` after converting it to a dataframe:
+`results.groupby('iter')['resource_iter'].unique()`
+
+Choosing a resource to budget
+-----------------------------
+
+By default, the budget is defined in terms of number of samples. That is,
+each iteration will use an increasing amount of samples to train on. You can
+however manually specify a parameter to use as the budget with the
+``resource`` parameter. Here is an example where the budget is defined in
+terms of the number of estimators of a random forest::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.model_selection import HalvingGridSearchCV
+    >>> import pandas as pd
+    >>>
+    >>> param_grid = {'max_depth': [3, 5, 10],
+    ...               'min_samples_split': [2, 5, 10]}
+    >>> base_estimator = RandomForestClassifier(random_state=0)
```

Comment:
wouldn't it make sense to use `warm_start=True` to reuse the trees that were previously grown in the previous iteration since `HalvingGridSearchCV` is called with `resource='n_estimators'`?

Reply:
Actually we probably need a test for this case to check that it's possible to incrementally build a random forest this way.

Reply:
Ok let's differ the `warm_start=True` optimization for later then.